lstgen/generators/go.py

```diff
@@ -0,0 +1,268 @@
+                for const in self.parser.constants:
+                    if const.comment is not None:
+                        wr.nl()
+                        self._write_comment(const.comment, False)
+
+                    if const.type.endswith('[]'):
+                        const.value = '[{}]'.format(const.value[1:-1])
+
+                    value = self.convert_to_go(const.value)
+                    #print("CONST", const.name, const.value, value)
+                    wr.writeln('{const.name}: {value},'.format(
+                        const=const,
+                        value=value
+                    ))
+
+                wr.nl()
+                wr.writeln("// ------------------------ Variables -------------------------")
+                for (comment, variables) in [
+                        ('Input variables', self.parser.input_vars),
+                        ('Output variables', self.parser.output_vars),
+                        ('Internal variables', self.parser.internal_vars),
+                    ]:
+
+                    for var in variables:
+                        if var.default is None:
+                            continue
+
+                        if var.comment is not None:
+                            wr.nl()
+                            self._write_comment(var.comment, False)
+
+                        value = self.convert_to_go(var.default)
+                        wr.writeln('{}: {},'.format(var.name, value))
+
+
+        # create setters for input vars
+        for var in self.parser.input_vars:
+            wr.nl()
+            signature = 'func (this *{cls}) Set{cap}(value {type})'.format(
+                cls=self.class_name,
+                cap=var.name.capitalize(),
+                type=self._convert_vartype(var.type)
+            )
+            with wr.indent(signature):
+                wr.writeln('this.{} = value'.format(var.name))
+
+        # create getters for output vars
+        for var in self.parser.output_vars:
+            wr.nl()
+            signature = 'func (this *{cls}) Get{cap}() {type}'.format(
+                cls=self.class_name,
+                cap=var.name.capitalize(),
+                type=self._convert_vartype(var.type)
+            )
+            with wr.indent(signature):
+                wr.writeln('return this.{}'.format(var.name))
+
+        self._write_method(self.parser.main_method)
+        for method in self.parser.methods:
+            self._write_method(method)
+        wr.nl()
+
+    def _convert_vartype(self, vartype):
+        return {
+            'BigDecimal.ZERO': 'decimal.NewFromInt(0)',
+            'BigDecimal[]': '[]'+self.bd_class,
+            'BigDecimal': ''+self.bd_class,
+            'int': 'int64',
+            'double': 'int64',
+        }[vartype]
+
+    def _conv_list(self, node):
+        res = super(GoGenerator, self)._conv_list(node)
+        return ['[]'+self.bd_class]+res
```

Comment:
PEP8, please add blanks on both sides of "+" operands:
`return ['[]' + self.bd_class] + res`